Composer/packages/client/src/components/Pagination/index.tsx

```diff
@@ -16,52 +17,52 @@ export interface IPaginationProps {
 }
 
 const createDropdownOption = (pageCount: number) => {
-  const options: IDropdownOption[] = [];
-  for (let i = 1; i <= pageCount; i++) {
-    options.push({ key: 'page' + i, text: '' + i });
-  }
-  return options;
+  return range(pageCount).map((_, i) => ({
+    key: `page ${i}`,
+    text: `${i}`,
+  }));
 };
 
 export const Pagination: React.FC<IPaginationProps> = (props) => {
-  const [index, setIndex] = useState(0);
+  const [index, setIndex] = useState<number>(0);
   const { pageCount, onChange } = props;
 
   useEffect(() => {
     setIndex(0);
   }, [pageCount]);
 
+  const changePage = (page: number) => {
+    setIndex(page);
+    onChange(page + 1);
+  };
+
   const handlePageSelected = (event: React.FormEvent<HTMLDivElement>, item?: IDropdownOption, index?: number) => {
-    setIndex(index || 0);
-    onChange(index ? index + 1 : 1);
+    if (index == null) return;
+    changePage(index);
   };
 
-  const hanglePreviousClick = () => {
-    const current = index - 1;
-    setIndex(current);
-    onChange(current + 1);
+  const handlePreviousClick = () => {
+    changePage(index - 1);
   };
 
-  const hangleNextClick = () => {
-    const current = index + 1;
-    setIndex(current);
-    onChange(current + 1);
+  const handleNextClick = () => {
+    changePage(index + 1);
```

Comment:
Consider using `useCallback` for these two handlers as they have a dependency on `index`.

Reply:
I don't think that would get us much. These handlers only fire if the index changes, so `useCallback` would just always go off. There's no way to take an action on the paginator that leaves the page unchanged except for the edge case of selecting it from the dropdown (and even them I'm not sure if the component fires its `onChange` if the same item gets selected as is already selected).